pypeit/core/extract.py

```diff
@@ -293,6 +293,32 @@ def extract_optimal(sciimg,ivar, mask, waveimg, skyimg, rn2_img, oprof, box_radi
     var_no = np.abs(skyimg - np.sqrt(2.0) * np.sqrt(rn2_img)) + rn2_img
 
     ispec, ispat = np.where(oprof > 0.0)
+
+    # Exit gracefully if we have no positive object profiles, since that means something was wrong with object fitting
+    if not np.any(oprof > 0.0):
+        junk = np.zeros(nspec)
```

Comment:
I'd like to talk through if there's a better way to do this.

Reply:
Happy to hear your suggestions. The point is that there are some objects very near the edges of the slit that are spuriously identified because our slit finding algorithm is too generous with the slit finding boundaries. This bit of code prevents a crash. The pipeline constantly crashes in weird problematics cases, whereas I think we need to make some effort to get it to exit gracefully when something is wrong.

Reply:
I'll need to think through how this is used to provide any substantive suggestions.

In terms of how you're using this, I'm really uncomfortable with code that can fault then exit quietly.  I would much prefer that it simply keep faulting until we either fix it or provide an informative exception string that explains how a user should proceed.  Quiet faults can be missed, whereas hard faults have to be dealt with.

Reply:
The code would not have these issues if the other parts of the code, i.e. slit finding, were more robust. I can also tune a parameter to more aggressively ignore edge objects that get identified. But in the end the issue is with our slit edges. I know how to fix this and will add that code in another PR.  Anyone looking at this data is going to realize an aperture full of zeros is problematic, and won't miss anything. By simply running the PypeIt twod spec script you will figure out what the issue is. When the code crashes, we are not able to diagnose other problems or get useful data. This is not a bug in the extraction algorithms, but rather sloppy coding elsewhere. That said, I could work harder to find these situations, and simply return a less sophisticated extraction.